exporter/googlecloudexporter/config.go

```diff
@@ -69,6 +73,51 @@ func (c *Config) getClientOptions() []option.ClientOption {
 	return opts
 }
 
+// setProject sets the project id from credentials if not already set
+func (c *Config) setProject() error {
+	if c.GCPConfig.Config.ProjectID != "" {
+		return nil
+	}
+
+	switch {
+	case c.Credentials != "":
+		return c.updateProjectFromJSON([]byte(c.Credentials))
+	case c.CredentialsFile != "":
+		return c.updateProjectFromFile(c.CredentialsFile)
+	default:
+		return nil
+	}
+}
+
+func (c *Config) updateProjectFromJSON(jsonBytes []byte) error {
+	jsonMap := make(map[string]interface{})
+	if err := json.Unmarshal(jsonBytes, &jsonMap); err != nil {
+		return fmt.Errorf("failed to unmarshal credentials: %w", err)
+	}
+
+	value, ok := jsonMap["project_id"]
+	if !ok {
+		return errors.New("project id does not exist")
+	}
+
+	strValue, ok := value.(string)
+	if !ok {
+		return errors.New("project id is not a string")
+	}
+
+	c.GCPConfig.ProjectID = strValue
+	return nil
+}
+
+func (c *Config) updateProjectFromFile(fileName string) error {
+	jsonBytes, err := ioutil.ReadFile(fileName)
```

Comment:
Nit: ioutil is deprecated, the functions were split between the os and io packages
```suggestion
	jsonBytes, err := os.ReadFile(fileName)
```